packages/stryker-jasmine-runner/test/unit/JasmineTestRunnerSpec.ts

```diff
@@ -0,0 +1,124 @@
+    expect(jasmineStub.env.throwOnExpectationFailure).calledWith(true);
+    expect(jasmineStub.exit).ok;
+    expect(jasmineStub.clearReporters).called;
+    expect(jasmineStub.randomizeTests).calledWith(false);
+  });
+
+  it('should clear require cache on run', async () => {
+    require.cache['foo.js'] = 'foo';
+    require.cache['bar.js'] = 'bar';
+    await actEmptyRun();
+    expect(require.cache['foo.js']).not.ok;
+    expect(require.cache['bar.js']).not.ok;
+  });
+
+  it('should report completed specs', async () => {
+    function addReporter(rep: jasmine.CustomReporter) {
+      rep.specDone({ id: 'spec0', fullName: 'foo spec', status: 'success' });
+      rep.specDone({ id: 'spec1', fullName: 'bar spec', status: 'failure', failedExpectations: [{ message: 'bar failed' }] });
+      rep.specDone({ id: 'spec2', fullName: 'disabled', status: 'disabled' });
+      rep.specDone({ id: 'spec3', fullName: 'pending', status: 'pending' });
+      rep.specDone({ id: 'spec4', fullName: 'excluded', status: 'excluded' });
+      rep.jasmineDone();
+    }
+    jasmineStub.addReporter.callsFake(addReporter);
+    const result = await sut.run({});
+    expect(result.status).eq(RunStatus.Complete);
+    expectTestResultsToEqual(result.tests, [
+      { name: 'foo spec', status: TestStatus.Success, failureMessages: undefined },
+      { name: 'bar spec', status: TestStatus.Failed, failureMessages: ['bar failed'] },
+      { name: 'disabled', status: TestStatus.Skipped, failureMessages: undefined },
+      { name: 'pending', status: TestStatus.Skipped, failureMessages: undefined },
+      { name: 'excluded', status: TestStatus.Skipped, failureMessages: undefined }
+    ]);
+  });
+
+  it('should time spec duration', async () => {
+    function addReporter(rep: jasmine.CustomReporter) {
+      rep.specStarted();
+      clock.tick(10);
+      rep.specDone({ fullName: 'foobar spec', id: 'spec0' });
+      rep.jasmineDone();
+    }
+    jasmineStub.addReporter.callsFake(addReporter);
+    const result = await sut.run({});
+    const expectedTestResult: TestResult = {
+      name: 'foobar spec',
+      status: TestStatus.Success,
+      timeSpentMs: 10,
+      failureMessages: undefined
+    };
+    expect(result.tests).deep.eq([expectedTestResult]);
+  });
+
+  it('should report errors on run', async () => {
+    const error = new Error('foobar');
+    jasmineStub.execute.throws(error);
+    const result = await sut.run({});
+    expect(result.status).eq(RunStatus.Error);
+    expect(result.errorMessages).lengthOf(1);
+    expect((result.errorMessages || [])[0]).matches(/An error occurred while loading your jasmine specs.*/)
+      .and.matches(/.*Error: foobar.*/);
+  });
+
+  it('should evaluate testHooks in global context', async () => {
+    const hooks = 'foobar';
+    await actEmptyRun(hooks);
+    expect(evalGlobalStub).calledWith(hooks);
+  });
+
+  function actEmptyRun(testHooks?: string) {
```

Comment:
Could you add some comments to describe what this does? (And in what way it's `empty`)

Reply:
Renamed to `actRunWithoutTests`